lib/preload.js

```diff
@@ -356,14 +392,18 @@ preload.run = (resin, docker, options) => {
   })
   .then((token) => {
     options.apiToken = token
-    return getContainerSize(options.apiHost, options.apiToken, options.registryHost, options.imageRepo)
+    return Promise.all([
+      getContainerSize(options.apiHost, options.apiToken, options.registryHost, options.imageRepo),
+      getPort()
+    ])
   })
-  .then((containerSize) => {
+  .spread((containerSize, dockerPort) => {
     options.containerSize = containerSize
+    options.dockerPort = dockerPort
     return createContainer(docker, options)
   })
   .then((container) => {
-    return runContainer(docker, container)
+    return runContainer(docker, container, options.appData.imageId, options.dockerPort)
```

Comment:
Same here, I think `runContainer( docker, container, options )` would be neater & more concise

Reply:
`runContainer` is no longer a good name for this.
I renamed it to `runPreloadContainerAndPull`.

I prefer having explicit parameters here as nothing is optional.